Firestore/src/Query.php

```diff
@@ -766,6 +712,27 @@ private function buildPosition($key, $fieldValues, $before)
         ], true);
     }
 
+    /**
+     * Find fieldFilters in a composite filter.
+     *
+     * @param array $filter The composite filter.
+     * @return array array of field filters.
+     */
+    private function flattenFilter($filter)
```

Comment:
JFYI, unlike C#, we are not removing the base `AND` composition if there's just one condition inside!

Reply:
We do remove that first level `and` in PHP-Firestore if not required. Please see the code here: https://github.com/googleapis/google-cloud-php/blob/main/Firestore/src/Query.php#L945